src/Symfony/Component/Routing/Loader/YamlFileLoader.php

```diff
@@ -152,29 +152,58 @@ protected function parseImport(RouteCollection $collection, array $config, $path
         $this->setCurrentDir(dirname($path));
 
         $subCollection = $this->import($config['resource'], $type, false, $file);
-        /* @var $subCollection RouteCollection */
-        $subCollection->addPrefix($prefix);
-        if (null !== $host) {
-            $subCollection->setHost($host);
-        }
-        if (null !== $condition) {
-            $subCollection->setCondition($condition);
-        }
-        if (null !== $schemes) {
-            $subCollection->setSchemes($schemes);
-        }
-        if (null !== $methods) {
-            $subCollection->setMethods($methods);
-        }
-        $subCollection->addDefaults($defaults);
-        $subCollection->addRequirements($requirements);
-        $subCollection->addOptions($options);
+        
+        if (is_array($subCollection)) {
+            /* @var $value RouteCollection */
+            foreach($subCollection as $value) {
```

Comment:
This creates a lot of duplicate code. What if you simply ensure it's always an array?

```php
if (!is_array($subCollection)) {
    $subCollection = array($subCollection);
}
```

Reply:
absolutely, I didnt know the best way to deal with, thanks you.